include/logger/logger_lib.hpp

```diff
@@ -1,9 +1,357 @@
 #ifndef LOGGER_LIB_HPP
 #define LOGGER_LIB_HPP
-
+#include "err.hpp"
 #include "sink.hpp"
+#include <vector>
+#include <exception>
+
+/*
+ *@brief Macro-Log logs the data at the specified level.
+ *@param level log level
+ *@param message logging message
+ */
+#define LOG(level, message) logger::log(level, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
+
+/*
+ *@brief Macro-TRACE logs the data at the trace level.
+ *@param message logging message
+ */
+#define TRACE(message) logger::log(logger::LEVEL::TRACE, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
+
+/*
+ *@brief Macro-DEBUG logs the data at the debug level.
+ *@param message logging message
+ */
+#define DEBUG(message) logger::log(logger::LEVEL::DEBUG, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
+
+/*
+ *@brief Macro-WARN logs the data at the warn level.
+ *@param message logging message
+ */
+#define WARN(message) logger::log(logger::LEVEL::WARN, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
+
+/*
+ *@brief Macro-ERROR logs the data at the error level.
+ *@param message logging message
+ */
+#define ERROR(message) logger::log(logger::LEVEL::ERROR, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
+
+/*
+ *@brief Macro-FATAL logs the data at the fatal level.
+ *@param message logging message
+ */
+#define FATAL(message) logger::log(logger::LEVEL::FATAL, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
 
+/*
+ *@brief Macro-OFF logs the data at the off level.
+ *@param message logging message
+ */
+#define OFF(message) logger::log(logger::LEVEL::OFF, message, __DATE__, __TIME__, __LINE__, __FILE__, __FUNCTION__)
+/*
+ *@brief Sets the user defined sink.
+ *@param sink_ptr pointer to user defined sink
+ */
+#define SET_SINK(sink_ptr) logger::set_sink(sink_ptr)
+
+/*
+ *@brief Set the log severity level.
+ *@param level desired severity level
+ */
+#define SET_SEVERITY_LEVEL(level) logger::set_severity_level(level)
+
+/*
+ *@brief Sets the logger name.
+ *@param logger_name logger name
+ */
+#define SET_LOGGER_NAME(logger_name) logger::set_logger_name(logger_name)
 namespace logger {
+/*
+ *@brief Sets the log severity level.
+ *@param level  desired log severity level
+ *@retval None
+ */
+void set_severity_level(logger::LEVEL level);
```

Comment:
can we take away All following global function if we inject logger inside macro ? 

I would like you to give a thought on this part of the code. All function are repeated  twice so take away the redundant code.